src/coreclr/jit/codegen.h

```diff
@@ -1056,8 +1056,8 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 #ifdef TARGET_ARM64
     void genCodeForConditionalCompare(GenTreeOp* tree, GenCondition prevCond);
     void genCodeForContainedCompareChain(GenTree* tree, bool* inchain, GenCondition* prevCond);
-    void genCodeForSelect(GenTreeConditional* tree);
 #endif
+    void genCodeForSelect(GenTreeOp* select);
```

Comment:
Why the switch to using GenTreeOp instead of GenTreeConditional ? Surely you'll always be passing a GenTreeConditional into it.

Reply:
The `GT_SELECT_HI` node is just a `GenTreeOp` since it only has two operands. We could make it a `GenTreeConditional` with a null `gtCond` as an alternative, but it would require changing the visitor code also. I don't have a super strong opinion on it.